core/schema/git.go

```diff
@@ -120,62 +125,141 @@ func (s *gitSchema) git(ctx context.Context, parent *core.Query, args gitArgs) (
 		if err != nil {
-			return nil, fmt.Errorf("failed to get client metadata from context: %w", err)
+			return nil, fmt.Errorf("failed to get client resource name: %w", err)
 		}
 
-		if clientMetadata != nil && clientMetadata.SSHAuthSocketPath != "" {
-			accessor, err := core.GetClientResourceAccessor(ctx, parent, clientMetadata.SSHAuthSocketPath)
-			if err != nil {
-				return nil, fmt.Errorf("failed to get client resource name: %w", err)
-			}
-
-			var sockInst dagql.Instance[*core.Socket]
-			if err := s.srv.Select(ctx, s.srv.Root(), &sockInst,
-				dagql.Selector{
-					Field: "host",
-				},
-				dagql.Selector{
-					Field: "__internalSocket",
-					Args: []dagql.NamedInput{
-						{
-							Name:  "accessor",
-							Value: dagql.NewString(accessor),
-						},
+		var sockInst dagql.Instance[*core.Socket]
+		if err := s.srv.Select(ctx, s.srv.Root(), &sockInst,
+			dagql.Selector{
+				Field: "host",
+			},
+			dagql.Selector{
+				Field: "__internalSocket",
+				Args: []dagql.NamedInput{
+					{
+						Name:  "accessor",
+						Value: dagql.NewString(accessor),
 					},
 				},
-			); err != nil {
-				return nil, fmt.Errorf("failed to select internal socket: %w", err)
-			}
+			},
+		); err != nil {
+			return nil, fmt.Errorf("failed to select internal socket: %w", err)
+		}
 
-			if err := socketStore.AddUnixSocket(sockInst.Self, clientMetadata.ClientID, clientMetadata.SSHAuthSocketPath); err != nil {
-				return nil, fmt.Errorf("failed to add unix socket to store: %w", err)
-			}
-			authSock = sockInst.Self
+		if err := socketStore.AddUnixSocket(sockInst.Self, clientMetadata.ClientID, clientMetadata.SSHAuthSocketPath); err != nil {
+			return nil, fmt.Errorf("failed to add unix socket to store: %w", err)
 		}
+		authSock = sockInst.Self
 	}
 
+	// 4. Handle git directory configuration
 	discardGitDir := false
 	if args.KeepGitDir != nil {
 		slog.Warn("The 'keepGitDir' argument is deprecated. Use `tree`'s `discardGitDir' instead.")
 		discardGitDir = !*args.KeepGitDir
 	}
 
+	// 5. Handle PAT auth
+	var authToken *core.Secret = nil
+	mainClientCallerID, err := parent.Server.MainClientCallerID(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve mainClientCallerID: %w", err)
+	}
+
+	// Isolate auth token to main client
+	if clientMetadata != nil && clientMetadata.ClientID == mainClientCallerID {
+		// extract scheme from ref
+		remote, err := gitutil.ParseURL(args.URL)
+		if errors.Is(err, gitutil.ErrUnknownProtocol) {
+			remote, err = gitutil.ParseURL("https://" + args.URL)
+		}
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse Git URL: %w", err)
+		}
+
+		if remote.Scheme == "https" || remote.Scheme == "http" {
+			// Check if repo is public
+			repo := git.NewRemote(memory.NewStorage(), &config.RemoteConfig{
+				Name: "origin",
+				URLs: []string{remote.Remote},
+			})
+
+			_, err := repo.List(&git.ListOptions{Auth: nil})
```

Comment:
nit: should probably use `ListContext` so we can cancel the request if the call is canceled